packages/graphql-hooks/src/GraphQLClient.js

```diff
@@ -219,8 +228,9 @@ class GraphQLClient {
         })
       })
       .then(result => {
-        if (result.error && this.logErrors) {
+        if (result.error) {
           this.logErrorResult({ result, operation })
+          this.notifyErrorResult({ result, operation })
         }
```

Comment:
Let's do the `if's` here, removing the responsibility from `logErrorResult` etc:

```js
if (result.error) {
  if (this.logErrors) {
    this.logErrorResult({ result, operation })
  }

  if (this.onError) {
    this.onError({ result, operation })
  }
}
```